Initial content

ferricoxide opened the pull request:
This PR includes logic for:

- Ensuring the Cribl-Edge software's dependencies installed
- Ensuring the Cribl-Edge software gets exctracted to the desired directory
- Sets the Cribl-Edge Run-mode
- Ensuring the Cribl-Edge software can log into /var/log
- Ensuring the Cribl-Edge software _does_ log into /var/log
- Sets the Cribl-Edge boot-mode to systemd
- Starts the Cribl-Edge systemd Unit

Presumably, will need to further add:

- `fapolicyd` logic (to support hardened-EL8 installations)
- `firewalld` exemptions (to allow remote management-node to contact the service-listener)
- Client-configuration so that the `cribl-edge` service knows how to reach out to its management node

The `firewalld` and "client-configuration" _may_ be mutually-exclusive. Still digging through documents to see what the node/manager relationship-requirements are. Will need information about the development environment's management-server to configure either or both.

ferricoxide committed:
Initial logic for downloading/placing cribl archive

ferricoxide committed:
Standard Jinja files from salt-formula template-repository

ferricoxide committed:
Illustrative Pillar content to support cribl-edge

ferricoxide committed:
Ensure cribl-edge logs into /var/log

ferricoxide committed:
Set cribl to edge-mode

ferricoxide committed:
Enable/start cribl-edge service

ferricoxide committed:
Add the starters

ferricoxide committed:
Enforce better dependency-chain

While we're here, make state-names more consistent

ferricoxide committed:
Fix lint-error - indent-padding

lorengordon commented on cribl-edge/config/log.sls:
```diff
@@ -0,0 +1,36 @@
+    - group: 'root'
+    - mode: '0700'
+    - name: '/var/log/cribl-edge'
+    - require:
+      - archive: 'Cribl-Edge Archive Extracted'
```
Will run into a bit of an ordering issue, if this state is called directly, e.g. `salt-call state.sls cribl-edge.config.log`. The way to work around that problem is to include the sls that defines this "archive" state. For an example, see this setup in the forescout formula:

https://github.com/plus3it/forescout-secure-connector-formula/blob/35705d6f49464956a79d67455c797c39d03a999f/forescout-secure-connector/service/running.sls#L3-L8

ferricoxide replied:
Ok, I _think_ we're good on this. I ran:

1. `salt-call -c /opt/watchmaker/salt/ state.apply cribl-edge`
1. `salt-call -c /opt/watchmaker/salt/ state.apply cribl-edge.clean`
1. `salt-call -c /opt/watchmaker/salt/ state.apply cribl-edge.config.log`

With no errors.

lorengordon commented on cribl-edge/service/running.sls:
```diff
@@ -0,0 +1,28 @@
+    - name: 'Cribl-Edge.service'
+    - require:
+      - cmd: 'Cribl-Edge Configure systemd Unit'
+    - unless:
+      - '[[ $( systemctl is-active cribl-edge.service ) == "active" ]]'
```
This _shouldn't_ be need. It is expected that `service.running` is already stateful.

lorengordon reviewed (commented):
I don't see a defaults.yaml? That might at least be valuable for the `package.install_path` option?

It would also be useful to add `clean` sls states, to make it easy to start from scratch.

ferricoxide committed:
Merge branch 'main' into InitialContent

ferricoxide committed:
Variablize installation-root reference

ferricoxide committed:
Add parameter-defaults

ferricoxide committed:
Default to pulling directly from Cribl's CDN

ferricoxide committed:
Fix case of systemd unit-name

ferricoxide commented:
> I don't see a defaults.yaml? That might at least be valuable for the `package.install_path` option?

Added and tested (the `source`/`source_hash` now point to the Cribl CDN)

> It would also be useful to add `clean` sls states, to make it easy to start from scratch.

Working that, next.

ferricoxide committed:
Add systemd service-name to defaults

...and consume the default value where appropriate

ferricoxide committed:
Add `real_log_dir` and `unit_file` parameters

- `real_log_dir`: the location where Cribl-Edge files are *actually*
  stored
- `unit_file`: the systemd unit definition-file for Cribl-Edge

While we're here, reorder things alphabetically

ferricoxide committed:
Add cleaners for systemd unit-files and units-hierarchy

ferricoxide committed:
Add cleaner for app's files

ferricoxide committed:
Add a 'clean' state

ferricoxide committed:
Remove superfluous 'unless' condition

ferricoxide committed:
Ensure we don't run into ordering-issues in non-default invocations

lorengordon commented on pillar.exampl.yml:
Filename should be `pillar.example.yml`....

ferricoxide replied:
Fixed

lorengordon commented on cribl-edge/config/log.sls:
```diff
@@ -0,0 +1,40 @@
+  file.directory:
+    - group: 'root'
+    - mode: '0700'
+    - name: '/var/log/cribl-edge'
```
```suggestion
    - name: '{{ cribl_edge.package.real_log_dir }}'
```

ferricoxide replied:
Fixed.

lorengordon commented on cribl-edge/config/log.sls:
```diff
@@ -0,0 +1,40 @@
+    - name: '{{ cribl_edge.package.install_path }}/log'
+    - require:
+      - file: 'Cribl-Edge Nuke Default Log-Dir'
+    - target: '/var/log/cribl-edge'
```
```suggestion
    - target: '{{ cribl_edge.package.real_log_dir }}'
```

ferricoxide replied:
Fixed.

lorengordon commented on pillar.exampl.yml:
```diff
@@ -0,0 +1,9 @@
+---
+cribl-edge:
+  lookup:
+    package:
+      archive:
+        source: https://FQDN/<PATH_TO_CRIBL-EDGE_ARCHIVE_FILE>.tgz
+        source_hash: source_hash_url_or_value
+      install_path: <FILESYSTEM_INSTALL_LOCATION>
```
Update with new options:

```suggestion
    package:
      archive:
        source: https://FQDN/<PATH_TO_CRIBL-EDGE_ARCHIVE_FILE>.tgz
        source_hash: source_hash_url_or_value
      install_path: <FILESYSTEM_INSTALL_LOCATION>
      real_log_dir: /var/log/cribl-edge

    service:
      name: cribl-edge.service
      unit_file: /etc/systemd/system/cribl-edge.service
```

ferricoxide replied:
Done.

lorengordon commented on cribl-edge/package/clean.sls:
```diff
@@ -0,0 +1,16 @@
+Cribl-Edge App-Dir Removed:
+  file.absent:
+    - name: {{cribl_edge.package.install_path }}
+    - onlyif:
+      - '[[ -d {{cribl_edge.package.install_path }} ]]'
```
I don't think the onlyif is needed for the clean right? That check was only needed when creating the logdir setup, to stop it from deleting the symlink. For this, the statefulness of `file.absent` should be enough?

```suggestion
```

ferricoxide replied:
Fixed

lorengordon reviewed (commented):
Looks pretty clean I think. The new setup is nice. Just a few comments to put a bow on it, and then ready to go

ferricoxide committed:
Fix spacing in Jinja var-refs

ferricoxide committed:
Change static-path to parameter-value

ferricoxide committed:
Fix file-name

ferricoxide committed:
Add missing parms/vals

ferricoxide committed:
Remove redundant `onlyif` stanza
